searchcore/src/vespa/searchcore/proton/docsummary/summarycompacttarget.h

```diff
@@ -31,6 +25,32 @@ class SummaryCompactTarget : public searchcorespi::IFlushTarget {
 
     FlushStats getLastFlushStats() const override { return _lastStats; }
     uint64_t getApproxBytesToWriteToDisk() const override { return 0; }
+protected:
+    SummaryGCTarget(const vespalib::string &, vespalib::Executor & summaryService, IDocumentStore & docStore);
+private:
+
+    virtual size_t getBloat(const IDocumentStore & docStore) const = 0;
+    virtual Task::UP create(IDocumentStore & docStore, FlushStats & stats, SerialNum currSerial) = 0;
+
+    vespalib::Executor      &_summaryService;
+    IDocumentStore & _docStore;
+    FlushStats               _lastStats;
+};
+
+class SummaryCompactBloatTarget : public SummaryGCTarget {
```

Comment:
Please add class description on what this compacts.

Reply:
Fixed